src/main/scala/org/broadinstitute/dsde/workbench/sam/openam/LdapAccessPolicyDAO.scala

```diff
@@ -52,6 +52,18 @@ class LdapAccessPolicyDAO(protected val ldapConnectionPool: LDAPConnectionPool,
     }
   }
 
+  private def unmarshalResource(results: Entry): Resource = {
+    val resourceTypeName = getAttribute(results, Attr.resourceType).getOrElse(throw new WorkbenchException(s"${Attr.resourceType} attribute missing"))
+    val resourceId = getAttribute(results, Attr.resourceId).getOrElse(throw new WorkbenchException(s"${Attr.resourceId} attribute missing"))
+    val authDomain = getAttributes(results, Attr.authDomain).getOrElse(Set.empty).map(g => WorkbenchGroupName(g)).toSet
+
+    Resource(ResourceTypeName(resourceTypeName), ResourceId(resourceId), authDomain)
```

Comment:
if you want to express the error in return type, you can do something like
```
  private def unmarshalResource(results: Entry): Either[String, Resource] = {
    for{
      resourceTypeName <- getAttribute(results, Attr.resourceType).toRight(s"${Attr.resourceType} attribute missing")
      resourceId <- getAttribute(results, Attr.resourceId).toRight(s"${Attr.resourceId} attribute missing")
    } yield {
      val authDomain =  getAttributes(results, Attr.authDomain).getOrElse(Set.empty).map(g => WorkbenchGroupName(g)).toSet
      Resource(ResourceTypeName(resourceTypeName), ResourceId(resourceId), authDomain)
    }
  }
```

And change caller to
```
  override def listResourcesConstrainedByGroup(groupId: WorkbenchGroupIdentity): Future[Set[Resource]] = for{
    res <- Future(ldapSearchStream(resourcesOu, SearchScope.SUB, Filter.createEqualityFilter(Attr.authDomain, groupId.toString))(unmarshalResource))
    r <- res.parSequence.fold(err => Future.failed(new WorkbenchException(err)), r => Future.successful(r.toSet))
  } yield r
```
(you'll need to import `cats.implicits._` for this. This will return all entries fail to unmarshall.